src/main/java/core/basesyntax/interfaces/FruitOperationImpl.java

```
@@ -0,0 +1,28 @@
+package core.basesyntax.interfaces;
+
+import core.basesyntax.FruitRecord;
+
+import java.util.Map;
+
+public class FruitOperationImpl implements FruitOperation {
+
+    @Override
+    public void process(FruitRecord record, Map<String, Integer> stock) {
+
+        stock.putIfAbsent(record.getFruit(), 0);
+        int current = stock.get(record.getFruit());
+
+        switch(record.getOperation()) {
+            case "b":
+                stock.put(record.getFruit(), record.getQuantity());
+                break;
+            case "s":
+            case "r":
+                stock.put(record.getFruit(), current + record.getQuantity());
+                break;
+            case "p":
+                stock.put(record.getFruit(), current - record.getQuantity());
+                break;
+        }
```

No default/unknown-operation handling. The code silently ignores unknown operation values because there is no default case. Either add a default that throws RuntimeException or (preferably) rely on a Strategy lookup that throws an informative exception when an operation has no handler configured. Tests and the checklist expect informative failure on invalid operation codes .